src/Business Foundation/App/NoSeries/src/Setup/NoSeriesLines.Page.al

```diff
@@ -0,0 +1,155 @@
+                    ApplicationArea = Basic, Suite;
+                    ToolTip = 'Specifies the size of the interval by which you would like to space the numbers in the number series.';
+
+                    trigger OnValidate()
+                    begin
+                        CurrPage.Update(true);
+                    end;
+                }
+                field("Allow Gaps in Nos."; Rec."Allow Gaps in Nos.")
+                {
+                    ApplicationArea = Basic, Suite;
+                    ToolTip = 'Specifies that a number assigned from the number series can later be deleted. This is practical for records, such as item cards and warehouse documents that, unlike financial transactions, can be deleted and cause gaps in the number sequence. This setting also means that new numbers will be generated and assigned in a faster, non-blocking way. NOTE: If an error occurs on a new record that will be assigned a number from such a number series when it is completed, the number in question will be lost, causing a gap in the sequence.';
+
+                    trigger OnValidate()
+                    begin
+                        CurrPage.Update(true);
+                    end;
+                }
+                field(Open; Rec.Open)
+                {
+                    ApplicationArea = Basic, Suite;
+                    ToolTip = 'Specifies whether the number series line is open. It is open until the last number in the series has been used.';
+                }
+            }
+        }
+        area(factboxes)
+        {
+            systempart(Control1900383207; Links)
+            {
+                ApplicationArea = RecordLinks;
+                Visible = false;
+            }
+            systempart(Control1905767507; Notes)
+            {
+                ApplicationArea = Notes;
+                Visible = false;
+            }
+        }
+    }
+
+    actions
+    {
+    }
+
+    trigger OnAfterGetCurrRecord()
+    var
+        NoSeriesMgt: Codeunit NoSeriesMgt;
+    begin
+        if Rec."Allow Gaps in Nos." then
+            Rec."Last No. Used" := NoSeriesMgt.GetLastNoUsed(Rec);
+    end;
+
+    trigger OnAfterGetRecord()
+    var
+        NoSeriesMgt: Codeunit NoSeriesMgt;
+    begin
+        if Rec."Allow Gaps in Nos." then
+            Rec."Last No. Used" := NoSeriesMgt.GetLastNoUsed(Rec);
+    end;
+
+    trigger OnNewRecord(BelowxRec: Boolean)
+    var
+        NoSeriesLine: Record "No. Series Line";
+    begin
+        NoSeriesLine.Copy(Rec);
+        if BelowxRec then
+            if NoSeriesLine.FindLast() then begin
+                Rec."Increment-by No." := NoSeriesLine."Increment-by No.";
+                Rec."Allow Gaps in Nos." := NoSeriesLine."Allow Gaps in Nos.";
+            end;
+    end;
+
+    trigger OnInsertRecord(BelowxRec: Boolean): Boolean
+    var
+        NoSeriesLine: Record "No. Series Line";
+    begin
+        if NoSeriesLine.Get(Rec."Series Code", Rec."Line No.") then begin
```

Comment:
I believe this would be an overuse of SetLoadFields. The table already doesn't contain a lot of data, hence loading the other fields will speed up later calls that may use those fields (since the table is cached on the server). I would focus on adding it to places where there is a lot of data being loaded.